tests_pytest/pytorch/trainable_infrastructure/test_annealing.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import torch
+import pytest
+
+from model_compression_toolkit.trainable_infrastructure.pytorch.annealing_schedulers import LinearAnnealingScheduler
+
+
+def test_linear_annealing():
+    scheduler = LinearAnnealingScheduler(t_start=10, t_end=35, initial_val=3.4, target_val=-1.6)
+    for t in [0, 9, 10]:
+        assert _isclose(scheduler(t), 3.4)
```

Comment:
is there a similar assertion function like in unittest.Assert..?

Reply:
I never really thought about it I just assume that the extra API provides functionality/simplifies things that are necessary for testing (for instance, there is a different "Assert___" (Equal, NotEqual, True, False, ...) so the condition is more declarative (if you use the API correctly not like we normally do). It simplifies assert conditions so understanding what is the expected behavior is easier.

Anyway, if there is no special mechanism and this is the common practice then ok